Yosemite/Yosemite/Tools/POS/POSOrderService.swift

```diff
@@ -0,0 +1,215 @@
+              taxes: [],
+              total: "",
+              totalTax: "",
+              attributes: [],
+              addOns: [],
+              parent: nil,
+              bundleConfiguration: [])
+    }
+
+    init(orderItem: OrderItem) {
+        self.itemID = orderItem.itemID
+        self.productID = orderItem.productID
+        self.quantity = orderItem.quantity
+    }
+}
+
+public protocol POSOrderServiceProtocol {
+    /// Syncs order based on the cart.
+    /// - Parameters:
+    ///   - cart: Cart with optional items (product & quantity).
+    ///   - order: Optional latest remotely synced order. Nil when syncing order for the first time.
+    ///   - allProducts: Necessary for removing existing order items with products that have been removed from the cart.
+    /// - Returns: Order from the remote sync.
+    func syncOrder(cart: [POSCartItem], order: POSOrder?, allProducts: [POSItem]) async throws -> POSOrder
+
+    /// Updates status of an order and syncs it
+    /// - Parameters:
+    ///   - posOrder: POS order.
+    ///   - status: New order status.
+    /// - Returns: Updated and synced POS order.
+    func updateOrderStatus(posOrder: POSOrder, status: OrderStatusEnum) async throws -> POSOrder
+
+    /// Creates WOO Order from POS Order.
+    /// - Parameters:
+    ///   - posOrder: POS order.
+    /// - Returns: Order created from posOrder data.
+    func order(from posOrder: POSOrder) -> Order
+}
+
+public final class POSOrderService: POSOrderServiceProtocol {
+    // MARK: - Properties
+
+    private let siteID: Int64
+    private let ordersRemote: OrdersRemote
+
+    // MARK: - Initialization
+
+    public convenience init(siteID: Int64, credentials: Credentials) {
+        self.init(siteID: siteID, network: AlamofireNetwork(credentials: credentials))
+    }
+
+    public init(siteID: Int64, network: Network) {
+        self.siteID = siteID
+        self.ordersRemote = OrdersRemote(network: network)
+    }
+
+    // MARK: - Protocol conformance
+
+    public func syncOrder(cart: [POSCartItem], order posOrder: POSOrder?, allProducts: [POSItem]) async throws -> POSOrder {
+        let initialOrder: Order = createInitialOrder(from: posOrder)
+
+        let order = updateOrder(initialOrder, cart: cart, allProducts: allProducts)
+        let syncedOrder: Order
+        if posOrder != nil {
+            syncedOrder = try await ordersRemote.updatePOSOrder(siteID: siteID, order: order, fields: [.items])
+        } else {
+            syncedOrder = try await ordersRemote.createPOSOrder(siteID: siteID, order: order, fields: [.items, .status])
+        }
+        return POSOrder(order: syncedOrder)
+    }
+
+    private func createInitialOrder(from posOrder: POSOrder?) -> Order {
+        if let posOrder {
+            return order(from: posOrder)
+        }
+        else {
+            //    TODO: handle WC version under 6.3 when auto-draft status is unavailable as in `NewOrderInitialStatusResolver`
```

Comment:
We can remove this comment since we're adding WC version 6.6+ to the eligibility check logic in https://github.com/woocommerce/woocommerce-ios/issues/12878.